hedera-node/hapi-utils/src/test/java/com/hedera/node/app/hapi/utils/CommonUtilsTest.java

```diff
@@ -111,5 +208,112 @@ void detectsOverflowInVariousCases() {
+                put(TokenFreezeAccount, new BodySetter<>(TokenFreezeAccountTransactionBody.class));
+                put(TokenUnfreezeAccount, new BodySetter<>(TokenUnfreezeAccountTransactionBody.class));
+                put(TokenGrantKycToAccount, new BodySetter<>(TokenGrantKycTransactionBody.class));
+                put(TokenRevokeKycFromAccount, new BodySetter<>(TokenRevokeKycTransactionBody.class));
+                put(TokenDelete, new BodySetter<>(TokenDeleteTransactionBody.class));
+                put(TokenUpdate, new BodySetter<>(TokenUpdateTransactionBody.class));
+                put(TokenMint, new BodySetter<>(TokenMintTransactionBody.class));
+                put(TokenBurn, new BodySetter<>(TokenBurnTransactionBody.class));
+                put(TokenAccountWipe, new BodySetter<>(TokenWipeAccountTransactionBody.class));
+                put(TokenAssociateToAccount, new BodySetter<>(TokenAssociateTransactionBody.class));
+                put(TokenDissociateFromAccount, new BodySetter<>(TokenDissociateTransactionBody.class));
+                put(TokenUnpause, new BodySetter<>(TokenUnpauseTransactionBody.class));
+                put(TokenPause, new BodySetter<>(TokenPauseTransactionBody.class));
+                put(ScheduleCreate, new BodySetter<>(ScheduleCreateTransactionBody.class));
+                put(ScheduleSign, new BodySetter<>(ScheduleSignTransactionBody.class));
+                put(ScheduleDelete, new BodySetter<>(ScheduleDeleteTransactionBody.class));
+                put(Freeze, new BodySetter<>(FreezeTransactionBody.class));
+                put(ConsensusCreateTopic, new BodySetter<>(ConsensusCreateTopicTransactionBody.class));
+                put(ConsensusUpdateTopic, new BodySetter<>(ConsensusUpdateTopicTransactionBody.class));
+                put(ConsensusDeleteTopic, new BodySetter<>(ConsensusDeleteTopicTransactionBody.class));
+                put(ConsensusSubmitMessage, new BodySetter<>(ConsensusSubmitMessageTransactionBody.class));
+                put(UncheckedSubmit, new BodySetter<>(UncheckedSubmitBody.class));
+                put(TokenFeeScheduleUpdate, new BodySetter<>(TokenFeeScheduleUpdateTransactionBody.class));
+                put(UtilPrng, new BodySetter<>(UtilPrngTransactionBody.class));
+                put(CryptoApproveAllowance, new BodySetter<>(CryptoApproveAllowanceTransactionBody.class));
+            }
+        };
+
+        setters.forEach((function, setter) -> {
+            final var txn = TransactionBody.newBuilder();
+            setter.setDefaultInstanceFor(txn);
+            try {
+                final var input = txn.build();
+                assertEquals(function, functionOf(input));
+            } catch (final UnknownHederaFunctionality uhf) {
+                throw new IllegalStateException(uhf);
+            }
+        });
+    }
+
     private static final ByteString NONSENSE = ByteString.copyFromUtf8("NONSENSE");
+
+    private static class BodySetter<T, B> {
```

Comment:
This test looks more complicated than the functionality you are testing, which is almost always a bad sign. Is it really necessary to do all of this?

Reply:
Ah, never mind. I just realized that you just moved the test from elsewhere. We should probably create an issue to redesign this test.

Reply:
Yes, I just moved these tests since I moved functionOf() call from MiscUtils to CommonUtils and changed it to switch expression.